### PR TITLE
Fix the use case when toolbar already has tools

### PR DESCRIPTION
**Pull request by oneeyeman1**
This PR fixes the ticket #13888.

The patch attached there fixes the case for the XRC, however it is still present in the core library.

Please review and apply.

Code was tested with the patch attached to the ticket itself and ran on OSX 10.13 compiled against minimal required of 10.9.

The sample is not modified and documentation changes is not needed.

Please review and apply.

### CODE REVIEW

**Comment by oneeyeman1** [comment]
@vadz,
Could you please restart Travis for this PR?
Also - is there anything left here to do?

Thx.

File: src/osx/carbon/frame.cpp [hunk 1]
```diff
@@ -301,6 +301,7 @@ void wxFrame::SetToolBar(wxToolBar *toolbar)
 #if wxOSX_USE_NATIVE_TOOLBAR
     if ( toolbar )
         toolbar->MacInstallNativeToolbar( true ) ;
+    toolbar->Realize();
```
Comment (MaartenBent) [inline src/osx/carbon/frame.cpp]: Can `toolbar` be `NULL`? The line above would suggest so.

**Review by MaartenBent — COMMENTED** [review]
You can avoid all the whitespace changes by returning early:
```c++
if( !dynamic_cast<wxFrame *>( GetParent() )->GetToolBar() )
    return  true;
```
Also, maybe you need to return `false` in this case?

File: src/osx/cocoa/toolbar.mm [hunk 2]
```diff
@@ -1170,160 +1170,162 @@ - (BOOL) isFlipped
 
 bool wxToolBar::Realize()
 {
-    if ( !wxToolBarBase::Realize() )
-        return false;
+    if( dynamic_cast<wxFrame *>( GetParent() )->GetToolBar() )
```
Comment (vadz) [inline src/osx/cocoa/toolbar.mm]: This is just wrong. First of all, dynamic cast may fail and the code would just crash.

Second, what's the point of doing it? I have no idea why is this test needed here.

**Comment by vadz** [comment]
@oneeyeman1 Your patch contains 3 lines of non-whitespace changes of which 2 introduce potential crashes. This can't be applied in the present state, please reopen after fixing this.

**Comment by oneeyeman1** [comment]
@vadz,
I will look at this one and all other opened one by me gradually fixing them in the next couple of days.

Thank you for the review.

**Pull request closed without merging**